### DISCUSSION TITLE
🐛 --repoter not working with `check` and `lint` command when lint from stdin

### DISCUSSION THREAD

**Discussion by axetroy** [biomejs/biome#4073] [Ideas]
### Environment information

```block
CLI:
  Version:                      1.9.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v16.20.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0

```

### What happened?

```bash
echo "var a = 123" | npx @biomejs/biome lint --reporter=json --stdin-file-path=test.js
```

output to stdout

```
var a = 123
stdin ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × The contents aren't fixed. Use the `--fix` flag to fix them.

```

### Expected result

`--reporter` should works

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Why should reporters work on stdin? That's by design.

**Comment by axetroy** [comment]
I don't know what the purpose of this design is.

But I think the output should be formatted by reporter.

For example, I got a scenario now. 

I want to lint a file. But since I have already read the file, I want to pass it in through stdin to reduce the extra file reads.

In addition: I have tried using js-api but it is not ready for production yet and missing a lot of feature

**Comment by ematipico** [comment]
I didn't understand why a JSON reporter is needed in stdin. I understand you have a use case, but I honestly didn't understand what this use case does. 

Also, you expect the `--reporter` to work, but you didn't explain what you expect to get from the output. At the moment, when you call a command without `--write`, the stdin mode return the original code.

**Comment by Conaclos** [comment]
Is there a particular reason to not make `--reporter` compatible with stdin + `--stdin-file-path`?

**Comment by ematipico** [comment]
If we decide to do so, we need to explain to users why we implemented and what's a valid use case. I'm not against implementing stuff, but we need to be good at justifying things so we don't regret them in the future. That's part of our philosophy

**Comment by axetroy** [comment]
> we need to explain to users why we implemented and what's a valid use case.

For keeping `--reporter` consistent. When this flag is specified, all output should be formatted by the reporter, without any exceptions.
